impress-org/givewp
# Refactor: add support for React 18+ createRoot

Resolves [GIVE-1108]

## Description



ReactDOM.render is no longer supported in React 18.  Since we load react from WordPress and WordPress 6.2 ships with version 18, we need to update our implementation to use the React 18 api of `createRoot`.

Currently GiveWP requires WP 6.3 so moving forward we don't need to support react 17 anymore.

More details here:
- https://react.dev/blog/2022/03/08/react-18-upgrade-guide#updates-to-client-rendering-apis
- https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-reactdom-render
- https://make.wordpress.org/core/2023/03/07/upgrading-to-react-18-and-common-pitfalls-of-concurrent-mode/
- https://make.wordpress.org/core/2024/06/07/preparation-for-react-19-upgrade/

<img width="1259" alt="Screenshot 2024-08-08 at 11 13 13 AM" src="https://github.com/user-attachments/assets/ee79725b-89f2-4640-83ea-cf3408704de8">

<img width="701" alt="Screenshot 2024-08-08 at 11 15 54 AM" src="https://github.com/user-attachments/assets/3742394b-9f96-4156-ae3e-8a31b984246b">


## Affects



All places were are using ReactDOM.render including:
- Donation list table
- Donation forms list table
- Donor list table
- Subscriptions list table
- Donation form block
- v3 Donation form
- v3 Donation form confirmation receipt
- Donor dashboard
- Logs
- Welcome banner
- Onboarding wizard
- Reports

## Visuals



N/A

## Testing Instructions



Load all screens (in affects) and make sure they work expected with no react warnings about createRoot in console.

## Pre-review Checklist






[GIVE-1108]: https://stellarwp.atlassian.net/browse/GIVE-1108?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
